web/html/src/manager/visualization/hierarchy.tsx

```diff
@@ -262,13 +263,131 @@ class Hierarchy extends React.Component {
       hurl = "reference/systems/visualization-menu.html";
     }
 
+    const formModel = this.state.selectedSystemGroups.reduce((result, item, index) => {
+      result[index] = item;
+      return result;
+    }, {});
+
     return (
       <TopPanel title={t(window.title)} helpUrl={hurl}>
-        <button className="toggle-filter-button" onClick={this.showFilters}>
-          {t((this.state.showFilters ? "Hide" : "Show") + " filters")}
+        <button className="toggle-filter-button" onClick={() => this.toggleFilters()}>
+          {this.state.showFilters ? t("Hide filters") : t("Show filters")}
           <i className={"fa fa-caret-" + (this.state.showFilters ? "up" : "down")} aria-hidden="true"></i>
         </button>
-        <div id="visualization-filter-wrapper"></div>
+        <div id="visualization-filter-wrapper">
```

Comment:
This is not actually new DOM, this is the DOM the old jQuery code created, lifted out and bound to React so we can use our own form components.